File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.cs
using System;
using System.Linq;
using System.Text;

public class Solution {
    public int solution(string s) {
         int answer = 0;

 char target = s[0];

 int equalCount = 0;
 int unequalCount = 0;

 StringBuilder sb = new StringBuilder();

 foreach (char c in s)
 {
     if(target == '\0') target = c;

     sb.Append(c);

     if (target == c)
         equalCount++;
     else
         unequalCount++;

     if(equalCount == unequalCount)
     {
         sb.Append(' ');
         target = '\0';
     }
 }

 answer = sb.ToString().Trim().Count(c => c == ' ') + 1;

return answer;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0

    alpha_1 = s[0]
    count_1, count_2 = 0, 0

    for i, c in enumerate(s):
        if alpha_1 == "":
            alpha_1 = c

        if alpha_1 == c:
            count_1 += 1
        else:
            count_2 += 1

        if count_1 == count_2:
            alpha_1 = ""
            count_1, count_2 = 0, 0
            answer += 1
    else:
        if count_1 != count_2:
            answer += 1

    return answer